funsor/ops/array.py

```diff
@@ -409,7 +409,7 @@ def stack(parts, dim=0):
     raise NotImplementedError
 
 
-stack.register(arraylist)(np.stack)
+stack.register(typing.Tuple[typing.Union[array + (int, float)], ...])(np.stack)
```

Comment:
Hmm, I wonder if there's a way to make this backend-agnostic, maybe
```py
stack.register(typing.Tuple[typing.Union[array], ...])(np.stack)
stack.register(typing.Tuple[typing.Union[int, float], ...])(make_array)
```
Anyway, I'll merge now since this is high priority, and we can support `backend==torch` later.